pkg/asset/installconfig/aws/validation.go

```diff
@@ -471,53 +475,40 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets Subnets, idxMap m
 
 func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) field.ErrorList {
 	allErrs := field.ErrorList{}
-	ec2Endpoint := ""
-	for id, service := range services {
-		err := validateEndpointAccessibility(service.URL)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Index(id).Child("url"), service.URL, err.Error()))
-			continue
-		}
-		if service.Name == ec2.ServiceName {
-			ec2Endpoint = service.URL
-		}
+	endpointOverrides := make(map[string]string)
+	for _, service := range services {
+		endpointOverrides[service.Name] = service.URL
 	}
 
-	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		if _, ok := partition.Regions()[region]; !ok && ec2Endpoint == "" {
-			err := validateRegion(region)
+	var errs []error
+	// validate the service endpoint for each of the required services. This will be the default endpoint for
+	// services where the user did not override the service endpoint through the install-config.
+	for _, service := range requiredServices {
+		if urlOverride, ok := endpointOverrides[service]; ok {
+			if err := validateEndpointAccessibility(urlOverride); err != nil {
+				errs = append(errs, field.Invalid(fldPath, region, fmt.Sprintf("failed to access %s endpoint at %s: %s", service, urlOverride, err.Error())))
+			}
+		} else {
+			defaultEndpoint, err := getDefaultServiceEndpoint(region, service, endpointOptions{
+				DisableHTTPS:         false,
+				UseDualStackEndpoint: awsv2.DualStackEndpointStateDisabled,
+			})
 			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, err.Error()))
+				errs = append(errs, field.Invalid(fldPath, region, fmt.Sprintf("could not determine %s endpoint for region %s: %s", service, region, err.Error())))
+			} else {
+				if err := validateEndpointAccessibility(defaultEndpoint); err != nil {
```

Comment:
I am a bit surprised that default Endpoints also have to be validated. Doing this additional validation seems purely precautionary.

Reply:
+1 I don't understand why we need to get/validate default endpoints.

Reply:
Looking at [the pr that added this validation](https://github.com/openshift/installer/pull/5432), it seems like the motivation for checking the endpoint resolution was to prevent long timeouts when an invalid region was entered. 

So the particular problem that was solving will be sensitive to at what point in the code it is running--i.e. i think it needs to be the first time the SDK session is used. I'm not sure if it makes sense to add this to the validate endpoints function.